client/tx/tx_test.go

```diff
@@ -306,25 +306,39 @@ func TestSign(t *testing.T) {
 		},
 
 		/**** test double sign Direct mode
-		  signing transaction with more than 2 signers should fail in DIRECT mode ****/
+		  signing transaction with 2 or more DIRECT signers should fail in DIRECT mode ****/
 		{
-			"direct: should fail to append a signature with different mode",
+			"direct: should append a DIRECT signature with existing AMINO",
+			// txb already has 1 AMINO signature
 			txfDirect, txb, from1, false,
-			[]cryptotypes.PubKey{},
+			[]cryptotypes.PubKey{pubKey2, pubKey1},
 			nil,
 		},
 		{
-			"direct: should fail to sign multi-signers tx",
+			"direct: should add single DIRECT sig in multi-signers tx",
 			txfDirect, txb2, from1, false,
+			[]cryptotypes.PubKey{pubKey1},
+			nil,
+		},
+		{
+			"direct: should fail to append 2nd DIRECT sig in multi-signers tx",
+			txfDirect, txb2, from2, false,
 			[]cryptotypes.PubKey{},
 			nil,
 		},
 		{
-			"direct: should fail to overwrite multi-signers tx",
-			txfDirect, txb2, from1, true,
+			"amino: should append 2nd AMINO sig in multi-signers tx with 1 DIRECT sig",
+			// txb2 already has 1 DIRECT signature
+			txfAmino, txb2, from2, false,
```

Comment:
Originally, there was a problem of failure when signing with SIGN_MODE_DIRECT when there is a signing with SIGN_MODE_AMINO in multi sign. This has been modified so that at least one person can sign with SIGN_MODE_DIRECT. Please ref https://github.com/cosmos/cosmos-sdk/pull/10683/files.